quora-api/src/main/java/com/upgrad/quora/api/controller/UserController.java

```diff
@@ -0,0 +1,47 @@
+package com.upgrad.quora.api.controller;
+
+import com.upgrad.quora.api.model.SignupUserRequest;
+import com.upgrad.quora.api.model.SignupUserResponse;
+import com.upgrad.quora.service.business.SignupBusinessService;
+import com.upgrad.quora.service.entity.UserEntity;
+import com.upgrad.quora.service.exception.SignUpRestrictedException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RestController;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("/")
+public class UserController {
+    final String ROLE_NON_ADMIN = "nonadmin";
+
+    @Autowired
+    SignupBusinessService signupBusinessService;
+
+    @RequestMapping(method = RequestMethod.POST, path = "user/signup", consumes = MediaType.APPLICATION_JSON_UTF8_VALUE, produces = MediaType.APPLICATION_JSON_UTF8_VALUE)
+    public ResponseEntity<SignupUserResponse> signup(final SignupUserRequest signupUserRequest) throws SignUpRestrictedException {
+        UserEntity userEntity = new UserEntity();
+        userEntity.setUuid(UUID.randomUUID().toString());
+        userEntity.setFirstName(signupUserRequest.getFirstName());
+        userEntity.setLastName(signupUserRequest.getLastName());
+        userEntity.setUserName(signupUserRequest.getUserName());
+        userEntity.setEmail(signupUserRequest.getEmailAddress());
+        userEntity.setPassword(signupUserRequest.getPassword());
+        userEntity.setSalt("salt");
```

Comment:
Do we need to set any default salt? The value will anyway get overwritten by the password encryption code. Please check and removed as needed

Reply:
Still open. Will resolve in the next commit